lib/util.js

```diff
@@ -420,6 +420,74 @@ exports.nextSleepTime = function (
     Math.min(base, previousSleep * 3));
 };
 
+
+/**
+ * Return next sleep time calculated by the jitter rule.
+ *
+ * @param {Number} numofRetries 
+ * @param {Number} currentSleepTime 
+ * @param {Number} maxSleepTime 
+ * @returns {JSON} return next sleep Time and totalTime.
+ */
+exports.jitteredSleepTime = function (numofRetries, currentSleepTime, totalTimeout, maxLoginTimeout) {
+  const nextSleepTime = Math.min(maxLoginTimeout - totalTimeout, 4 * numofRetries);
+  const sleep = nextSleepTime + getJitter(currentSleepTime);
+  totalTimeout += sleep
+
+  return {sleep, totalTimeout}
+}
+
+/**
+ * Choose one of the number between two numbers.
+ *
+ * @param {Number} firstNumber 
+ * @param {Number} secondNumber 
+ * @returns {Boolean} return a random number between two numbers.
+ */
+function chooseRandom (firstNumber, secondNumber) {
+  let bigNumber;
+  let smallNumber;
+  if (firstNumber > secondNumber) {
```

Comment:
WDYT about using Math.min() and Math.max()? Both min and max should be included or not?

Reply:
Fixed.